### ISSUE TITLE
JS Daylight Saving Date Difference

### ISSUE THREAD

**Issue by cscrudder** [Microsoft/ChakraCore#798]
**IE browser v:** 11.0.9600.18230
**OS:** 7 Enterprise
**Description:** When attempting to calculate the difference of time (hours) between two JavaScript date variables falling on either side of daylight saving time (spring forward / UTC Central) 2am is improperly handled.

Daylight saving time should recognize 2am (02:00:00) as 3am for spring forward. IE is currently inconsistent as other major browsers (who observe 2am on daylight saving time as 3am) as well as other programming languages (C# to be included).

**Sample code:**
```
var beginDateTime = new Date("03/13/2016 00:00:00");
var endDateTime_1 = new Date("03/13/2016 01:00:00");
var endDateTime_2 = new Date("03/13/2016 02:00:00");
var endDateTime_3 = new Date("03/13/2016 03:00:00");

alert("1 am: "+Math.abs(endDateTime_1 - beginDateTime) / 36e5 + " Hours");
alert("2 am: "+Math.abs(endDateTime_2 - beginDateTime) / 36e5 + " Hours");
alert("3 am: "+Math.abs(endDateTime_3 - beginDateTime) / 36e5 + " Hours");
```

**Comment by ljharb** [comment]
Please note that daylight savings time transition is locale-dependent - in Brazil, for example, it transitions at midnight.

**Comment by cscrudder** [comment]
@ljharb yes I am aware of the locale dependency, so the example may need to be tweaked to replicate the same issues.  The locale where the issue is apparent to me is in UTC Central.

However this does not change the fact that the hour in question to be 'forwarded' is not correctly observed/handled to be in uniform of other browser/languages.

**Comment by cscrudder** [comment]
With further investigation with a co-worker, we have identified this issue appears to be resolved in IE 11.162.10586.0
